src/afl-analyze.c

```diff
@@ -982,18 +934,27 @@ int main(int argc, char **argv, char **envp) {
 
   if (optind == argc || !in_file) usage(argv[0]);
 
+  if ((ptr = getenv("AFL_MAP_SIZE")) || (ptr = getenv("AFL_MAPSIZE"))) {
+
+    map_size = atoi(ptr);
+    if (map_size < 8 || map_size > (1 << 29))
+      FATAL("illegal AFL_MAP_SIZE %u, must be between 2^3 and 2^30", map_size);
+    if (map_size % 8) map_size = (((map_size >> 3) + 1) << 3);
+
+  }
+
   use_hex_offsets = !!get_afl_env("AFL_ANALYZE_HEX");
 
   check_environment_vars(envp);
 
   sharedmem_t shm = {0};
-  trace_bits = afl_shm_init(&shm, MAP_SIZE, 0);
+  trace_bits = afl_shm_init(&shm, map_size, 0);
```

Comment:
The initial idea was to keep track of the current and the actual map size in `afl_shm` - there are variables for both, namely:
https://github.com/AFLplusplus/AFLplusplus/blob/504529c3aa5c80937f9f722f90d0ec55e09c6dd2/include/sharedmem.h#L47
https://github.com/AFLplusplus/AFLplusplus/blob/504529c3aa5c80937f9f722f90d0ec55e09c6dd2/include/sharedmem.h#L48
Now parts of it are in the afl state/global, parts in the forkserver state (`fsrv.map_size`), yet the vars in the shared map still exist. Not blocking for this pr but we may want to clean up 2 of the 4 vars afterwards. 
I still feel like they belong close to the shared mem struct, but it's fine either way

Reply:
you are right. the map_size should be in sharedmem (and I renamed it to that to be obvious what it is and the name it is used everywhere else). the used size should be in the forkserver though as this is something the forkserver learns IMHO. removed the others

Reply:
new change ok?

Reply:
Yes thx :)